test/tests.ts

```diff
@@ -0,0 +1,99 @@
+import { assert } from "chai";
+import { readJson, unlink, writeJson } from "fs-extra";
+
+export async function expectErrorAsync(
+  f: () => Promise<any>,
+  errorMessage?: string
+) {
+  try {
+    await f();
+  } catch (err) {
+    assert.equal(err.message, errorMessage);
+  }
+}
+
+describe("Solhint plugin", function() {
+  const SOLHINT_CONFIG_FILENAME = ".solhint.json";
+
+  describe("Project with solhint config", function() {
+    before("setup", async function() {
+      process.chdir(__dirname + "/buidler-project");
+      process.env.BUIDLER_NETWORK = "develop";
+
+      delete require.cache[require.resolve("@nomiclabs/buidler")];
+      this.env = require("@nomiclabs/buidler");
+    });
+
+    it("should define solhint task", function() {
+      assert.isDefined(this.env.tasks.solhint);
+      assert.isDefined(this.env.tasks.check);
+    });
+
+    it("return a report", async function() {
+      const reports = await this.env.run("solhint");
+      assert.equal(reports.length, 1);
+      assert.equal(reports[0].reports.length, 5);
+    });
+  });
+
+  describe("Project with no solhint config", function() {
+    before("setup", function() {
+      process.chdir(__dirname + "/no-config-project");
+      process.env.BUIDLER_NETWORK = "develop";
+
+      delete require.cache[require.resolve("@nomiclabs/buidler")];
+      this.env = require("@nomiclabs/buidler");
+    });
+
+    it("should define solhint task", function() {
+      assert.isDefined(this.env.tasks.solhint);
+      assert.isDefined(this.env.tasks.check);
+    });
+
+    it("return a report", async function() {
+      const reports = await this.env.run("solhint");
+      assert.equal(reports.length, 1);
+      assert.equal(reports[0].reports[0].ruleId, "max-line-length");
+    });
+  });
+
+  describe("Project with invalid solhint configs", function() {
```

Comment:
These tests are good :) The invalid case I mentioned before, is when `solhint` can't even parser the file.